test/e2e/streaming_test.go

```diff
@@ -39,7 +40,9 @@ func (suite *StreamingTestSuite) SetupSuite() {
 
 func (suite *StreamingTestSuite) TearDownSuite() {
 	log.Info("Entering TearDownSuite()")
-	ctx.Cleanup()
+	if !debugMode || !t.Failed() {
```

Comment:
This is repeated all over, perhaps a `func handleCleanup(t *testing.T)` could be used instead?